worlds/civ_6/Items.py

```diff
@@ -0,0 +1,300 @@
+from enum import Enum
+from typing import Dict, Optional, TYPE_CHECKING, List, Union
+from BaseClasses import Item, ItemClassification
+from .Data import GoodyHutRewardData, get_era_required_items_data, get_existing_civics_data, get_existing_techs_data, get_goody_hut_rewards_data, get_progressive_districts_data
+from .Enum import CivVICheckType, EraType
+from .ProgressiveDistricts import get_flat_progressive_districts
+if TYPE_CHECKING:
+    from . import CivVIWorld
+CIV_VI_AP_ITEM_ID_BASE = 5041000
+
+NON_PROGRESSION_DISTRICTS = [
+    "PROGRESSIVE_PRESERVE",
+    "PROGRESSIVE_NEIGHBORHOOD"
+]
+
+
+# Items required as progression for boostsanity mode
+BOOSTSANITY_PROGRESSION_ITEMS = [
+    "TECH_THE_WHEEL",
+    "TECH_MASONRY",
+    "TECH_ARCHERY",
+    "TECH_ENGINEERING",
+    "TECH_CONSTRUCTION",
+    "TECH_GUNPOWDER",
+    "TECH_MACHINERY",
+    "TECH_SIEGE_TACTICS",
+    "TECH_STIRRUPS",
+    "TECH_ASTRONOMY",
+    "TECH_BALLISTICS",
+    "TECH_STEAM_POWER",
+    "TECH_SANITATION",
+    "TECH_COMPUTERS",
+    "TECH_COMBUSTION",
+    "TECH_TELECOMMUNICATIONS",
+    "TECH_ROBOTICS",
+    "CIVIC_FEUDALISM",
+    "CIVIC_GUILDS",
+    "CIVIC_THE_ENLIGHTENMENT",
+    "CIVIC_MERCANTILISM",
+    "CIVIC_CONSERVATION",
+    "CIVIC_CIVIL_SERVICE",
+    "CIVIC_GLOBALIZATION",
+    "CIVIC_COLD_WAR",
+    "CIVIC_URBANIZATION",
+    "CIVIC_NATIONALISM",
+    "CIVIC_MOBILIZATION",
+    "PROGRESSIVE_NEIGHBORHOOD",
+    "PROGRESSIVE_PRESERVE"
+]
+
+
+class FillerItemRarity(Enum):
+    COMMON = "COMMON"
+    UNCOMMON = "UNCOMMON"
+    RARE = "RARE"
+
+
+FILLER_DISTRIBUTION: Dict[FillerItemRarity, float] = {
+    FillerItemRarity.RARE: 0.025,
+    FillerItemRarity.UNCOMMON: 0.2,
+    FillerItemRarity.COMMON: 0.775,
+}
+
+
+class FillerItemData:
+    name: str
+    type: str
+    rarity: FillerItemRarity
+    civ_name: str
+
+    def __init__(self, data: GoodyHutRewardData):
+        self.name = data["Name"]
+        self.rarity = FillerItemRarity(data["Rarity"])
+        self.civ_name = data["Type"]
+
+
+def get_filler_item_data() -> Dict[str, FillerItemData]:
+    """
+    Returns a dictionary of filler items with their data
+    """
+    goody_huts = get_goody_hut_rewards_data()
+    # Create a FillerItemData object for each item
+    cached_filler_items = {item["Name"]: FillerItemData(item) for item in goody_huts}
+
+    return cached_filler_items
+
+
+class CivVIItemData:
+    civ_vi_id: Union[int, str]
+    classification: ItemClassification
+    name: str
+    code: int
+    cost: int
+    item_type: CivVICheckType
+    progression_name: Optional[str]
+    civ_name: Optional[str]
+    era: Optional[EraType]
+
+    def __init__(self, name: str, civ_vi_id: int, cost: int, item_type: CivVICheckType, id_offset: int, classification: ItemClassification, progression_name: Optional[str], civ_name: Optional[str] = None, era: Optional[EraType] = None):

```

Comment:
Ehh I'm just going to rework how this is handled so the civ_id will always be an int. It is all to support sending goody huts, but if I add an exception to the function rather than the base class that'll feel more appropriate

Reply:
I changed this:
```py
 elif item.item_type == CivVICheckType.GOODY and item_to_send.civ_name:                
                    item_to_send.civ_vi_id = item_to_send.civ_name
                    await ctx.game_interface.give_item_to_player(item_to_send, sender)
```
To this:
```py
elif item.item_type == CivVICheckType.GOODY and item_to_send.civ_name:
                    await ctx.game_interface.give_item_to_player(item_to_send, sender, game_id_override=item_to_send.civ_name)
```
It provides room for the edge case without requiring me to change the id (something that feels like should never be changed tbh)